### PR TITLE
BlockChain: use Network in constructors

### PR DESCRIPTION
**Pull request by msgilligan**
Constructors that take `NetworkParameters` are marked as `@VisibleForTesting`. This is because of the special `UNITTEST` `NetworkParameters` that is required for certain tests.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java [hunk 1]
```diff
@@ -160,13 +162,26 @@ static class OrphanBlock {
 
     private final VersionTally versionTally;
 
+    /**
+     * Constructs a BlockChain connected to the given list of listeners (wallets) and a store.
+     * @param network network parameters for this chain
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:167]: Is "parameters" accurate?

**Review by schildbach — COMMENTED** [review]
Looks good, except for 1 minor comment.

File: core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java]: Fixed and force-pushed.

**Comment by msgilligan** [comment]
> Looks good, except for 1 minor comment.

I fixed the comment text and force-pushed. I also removed two of the `@VisibleForTesting` constructors in `Blockchain` because they weren't being used and didn't seem necessary.

**Pull request closed without merging**